### DISCUSSION TITLE
AbnormalControllerState regularly firing

### DISCUSSION THREAD

**Discussion by hknlof** [strimzi/strimzi-kafka-operator#7081] [Q&A] [answered]
Hi Team,

in our alerting `AbnormalControllerState` is regularly firing. It would be great to know. As far as I understand strimzi is spinning up the pods and hence should also decide, whether a given broker is a controller as well.

Refs:
https://github.com/strimzi/strimzi-kafka-operator/blob/main/packaging/examples/metrics/prometheus-install/prometheus-rules.yaml#L28

**Comment by scholzj** [comment, marked as answer]
No, Kafka decides which broker is a controller. Strimzi is not involved in any way in that. Having multiple controllers might for example be caused by some network issues causing partitions between the nodes etc.

**Reply by hknlof** [reply]
Any recommendations on how to get it under control?

**Reply by scholzj** [reply]
TBH, I don't know. The logs should normally show some more details about the controller election, about any issues with controller co nectivity etc.

**Reply by hknlof** [reply]
I'll check and see what I can do thanks.
